### PR TITLE
replace words in script text

### PR DESCRIPTION
**Pull request by nashousni**
use ctrl +R key combination to replace words orccurences in a script text

### CODE REVIEW

File: gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.ResourceBundle;
+
+/**
+ * @author Nassirou Nambiema <nassirou.nambiena at rte-france.com>
+ */
+public class ReplaceWordBar extends HBox {
+
+    private final Button replaceButton;
+    private final Button replaceAllButton;
+    private final CustomTextField searchField = (CustomTextField) TextFields.createClearableTextField();
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java]: What an ugly cast...

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java [hunk 2]
```diff
@@ -73,25 +81,47 @@ public GroovyCodeEditor(Scene scene) {
...
+                setShowDetailNode(false);
+                if (!vBox.getChildren().contains(replaceWordBar)) {
+                    vBox.getChildren().add(replaceWordBar);
+                }
+                setDetailNode(vBox);
+                resetDividerPosition();
+                replaceWordBar.getReplaceAllButton().disableProperty().bind(validateDisableProperty());
+                replaceWordBar.getReplaceButton().disableProperty().bind(validateDisableProperty());
+                replaceWordBar.getReplaceAllButton().setOnAction(event -> replaceAllOccurences(searchBar.getSearchField().getText(), codeArea.getText()));
+                replaceWordBar.getReplaceButton().setOnAction(event -> replaceCurrentOccurence(searchBar.getMatcherCurrentMatchStart(), searchBar.getMatcherCurrentMatchEnd()));
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: There is a design issue here with the encapsulation principle: you should not be able to disable the buttons like this. This should be done internally in the replaceBar

File: gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        return replaceButton;
+    }
+
+    public Button getReplaceAllButton() {
+        return replaceAllButton;
+    }
+
+    public CustomTextField getSearchField() {
+        return searchField;
+    }
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java]: There is design issue here with the encapsulation principle: don’t give access to the components but provide business method: getSearchedText for instance

File: gse-util/src/main/java/com/powsybl/gse/util/SearchBar.java [hunk 4]
```diff
@@ -219,6 +219,50 @@ public SearchBar(Searchable textArea) {
...
+        return downButton;
+    }
+
+    public Button getUpButton() {
+        return upButton;
+    }
+
+    public SearchMatcher getMatcher() {
+        return matcher;
+    }
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/SearchBar.java]: There is an encapsulation issue here: don’t give access to all members but provide business method

Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/SearchBar.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java (hunk 3 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/ReplaceWordBar.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java (hunk 2 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GroovyCodeEditor.java]: updated

**Pull request merged**